src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -257,52 +257,100 @@ import scala.runtime.Statics
   *                        on the result (e.g. calling `head` or `tail`, or checking if it is empty).
   *  @define evaluatesAllElements This method evaluates all elements of the collection.
   */
-@SerialVersionUID(3L)
-final class LazyList[+A] private(private[this] var lazyState: () => LazyList.State[A])
+@SerialVersionUID(4L)
```

Comment:
I think this may cause `InvalidClassException` when we send/receive it across the network.
Would be better to have a test case with the old version hex dump to test if it works between the old and new, eg rolling update

Reply:
it specifically does not serialize across versions. there's really no way to allow it to do so